packages/ember-glimmer/tests/integration/helpers/yield-test.js

```diff
@@ -0,0 +1,256 @@
+import { RenderingTest, moduleFor } from '../../utils/test-case';
+import { set } from 'ember-metal/property_set';
+import { Component } from '../../utils/helpers';
+
+moduleFor('Helpers test: {{yield}} helper', class extends RenderingTest {
+
+  ['@test a component with a layout set renders its template where the {{yield}} helper appears']() {
+    this.registerComponent('yield-comp', { template: '<div class="wrapper">{{yield}}</div>' });
+
+    this.render('{{#yield-comp}}<div class="page-body">page body</div>{{/yield-comp}}', { title: 'test' });
+
+    equal(this.$('div.wrapper div.page-body').length, 1, 'page-body is embedded within wrapping my-page');
+
+    this.assertStableRerender();
+  }
+
+  ['@test block should work properly even when templates are not hard-coded']() {
+    this.registerComponent('outer-comp', { template: '<div class="wrapper">{{yield}}</div>' });
+    this.registerComponent('inner-comp', { template: '{{#outer-comp}}<div class="page-body">Body</div>{{/outer-comp}}' });
+
+    this.render('{{inner-comp}}');
+
+    equal(this.$('div.wrapper div.page-body').length, 1, 'page-body is embedded within wrapping my-page');
+
+    this.assertStableRerender();
+  }
+
+  ['@test templates should yield to block, when the yield is embedded in a hierarchy of virtual views']() {
+    let list = [1, 2, 3, 4, 5];
```

Comment:
This title is referring to a very old implementation strategy. I think it's just testing `{{yield}}` inside of `{{#each}}` or `{{#if}}`. I think each of those deserves its own test, with complete INUR coverage.